### PR TITLE
fix: #1244 remove ```currentConvoProvider```

### PR DESCRIPTION
**Pull request by gtalha07**
### PR Changes:
This essentially removes ```currentConvoProvider``` from our state system and instead goes with the approach of providing data to the riverpod convo dependencies through the use of ```typedef``` feature available in dart 3 which serves as alias for combining different type objects, thereby making it easier to pass as argument to family providers .i.e. they are limited to take only one type argument. To summarize the changes:

1. ```currentConvoProvider``` is removed whose state is ambiguous and instead pass ```typedef``` which is a ```Record``` in dart term and takes multiple type objects.
2.  Make convo providers take required arguments in hierarchal manner instead of watching global convo variable.

### CODE REVIEW

File: app/lib/common/utils/utils.dart [hunk 1]
```diff
@@ -290,3 +291,8 @@ const List<(RSVP, String)> rsvpOptions = <(RSVP, String)>[
   (RSVP.Maybe, 'Maybe'),
   (RSVP.Yes, 'Yes'),
 ];
+
```
Comment (gnunicorn) [inline app/lib/common/utils/utils.dart:294]: uhh... I like that. Very nice!

File: app/macos/Podfile.lock [hunk 2]
```diff
@@ -149,4 +149,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ce7dbe26c78bfc7ba46736094c1e2d25982870fa
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.12.1
```
Comment (gnunicorn) [inline app/macos/Podfile.lock:152]: we really like messing with this file ...

**Review by gnunicorn — APPROVED** [review]
Thanks heaps. This is great. Just some minor inquiries and questions from my side.

**Review by kumarpalsinh25 — APPROVED** [review]
Much needed code refactor. Great work!

File: app/lib/common/utils/utils.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/common/utils/utils.dart:294]: could place them in providers folder but looks cleaner here. Should move it?

File: app/macos/Podfile.lock (hunk 2 above)
Comment (gtalha07) [reply app/macos/Podfile.lock:152]: I am getting frustrated with this, the thing is it gets updated when you run it on your local for macos/ios platform. Can we add it in ```.gitignore``` list.

**Review by bitfriend — APPROVED** [review]
Great work

**Pull request merged**